XIPs/xip-universal-consent-layer.md

```diff
@@ -0,0 +1,55 @@
+# XIP-XX: Universal Consent Layer for XMTP
+
+| XIP                |                                            |
+|--------------------|--------------------------------------------|
+| Title              | Universal Consent Layer for XMTP           |
+| Author             | Pol Maire pol@getconverse.app              |
+| Status             | Draft                                      |
+| Type               | Standards Track                            |
+| Created            | 2023-04-08                                 |
+
+## Abstract
+
+This XIP proposes adding a universal consent layer to the XMTP protocol to prevent spam and allow users to maintain their communication preferences across applications.
+
+## Motivation
+
+Currently, the XMTP network allows anyone to write to anyone without the option to opt-in or opt-out of conversations with other users. Implementing a universal consent layer will provide users with more control over their communication preferences, reduce spam, and enhance privacy.
+
+## Specification
+
+### Consent layers
+
+1. **Opted-in**: The user has either sent at least one message in a conversation with the other user or explicitly accepted talking with the other user.
```

Comment:
One concern here is the number of messages needed to determine whether the user has sent a message in the convo. If clients are expected to read through the message history of every conversation, that is a lot of overhead (particularly for mobile apps).

If the most recent messages in the convo are all from the other party, the client would have to read through all of them until they found one from the user.

Reply:
@neekolas 100% agree!
We propose that this first version of the XIP only deals with explicit opt-in and opt-out.
Let's remove implicit opt-in (which is hard for the performance reason you just mentioned) 
We'll also add a section describing how an implicit opt-in should look like in clients that want to implement it (i.e. clients that sync all messages)